Automattic/woocommerce-payments
# Refund transaction from details page

Fixes #7248 

#### Changes proposed in this Pull Request

Adds a proxy endpoint to call server's refund endpoint directly from transaction details page bypassing.

- If there is no associated order for the transaction -> display a notice with the option to do a full refund. This option proxies directly to Stripe, bypassing WC's built-in refunding logic since there's no order to add any notes.
- Dropdown menu to do a full refund. If we have associated order, then we'll call WC build-in refund logic and will add notes to order
- Dropwon menu option to do a partial refund. Redirects to "edit order" page to perform partial refund.

#### Testing instructions

- Make payment and check it's details page
- Choose "Refund in full"
- The order and transaction should be refunded

- Make payment and check it's details page
- Choose "Partial refund"
- Assert you can do partial refund from "edit order" page

- Make payment and check it's details page
- add `delete charge.order` at the top of `PaymentDetailsSummary` component to simulate missing order case.
- Assert that there's no option to do a partial refund
- Assert that there's "missing order" notice
- Assert can refund from notice and dropdown.

-------------------


**Post merge**
